File: src/dataset/cpp/armstrong-numbers_exercism-chrisnp/claude/ClaudeSonnet4_armstrong_numbers_v2.cpp
#include "armstrong_numbers.h"

bool armstrong_numbers::is_armstrong_number(const int &num) noexcept {
    if (num < 0) return false;
    
    int temp = num;
    int num_digits = 0;
    
    // Count digits without string conversion
    if (temp == 0) {
        num_digits = 1;
    } else {
        while (temp > 0) {
            temp /= 10;
            ++num_digits;
        }
    }
    
    // Calculate sum using integer arithmetic
    temp = num;
    int sum_digits_raised = 0;
    while (temp > 0) {
        int digit = temp % 10;
        int power = 1;
        for (int i = 0; i < num_digits; ++i) {
            power *= digit;
        }
        sum_digits_raised += power;
        temp /= 10;
    }
    
    return sum_digits_raised == num;
}

File: src/dataset/javascript/armstrong-numbers_exercism-SLIB53/claude/ClaudeSonnet4_armstrong-numbers_v2.js
export const validate = integer => {
  const str = integer.toString();
  const length = str.length;
  let sum = 0;
  
  for (let i = 0; i < length; i++) {
    const digit = str.charCodeAt(i) - 48; // Convert char to digit
    sum += digit ** length;
  }
  
  return integer === sum;
};